lib/tests/streamlit/typing/slider_types.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+# Perform some "type checking testing"; mypy should flag any assignments that are incorrect.
+# Note: Due to https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html, mypy wil not detect
+# an <int> value being assigned to a <float> variable. There's nothing we can do about this, apparently.
+if TYPE_CHECKING:
+    from datetime import date, datetime, time, timedelta
+
+    from streamlit.elements.widgets.slider import SliderMixin
+
+    slider = SliderMixin().slider
+
+    singularint1: int = slider("foo", 5)
+    multiint1: tuple[int, int] = slider("foo", 5, 10)
+    singularfloat1: float = slider("foo", 5.0)
+    multifloat1: tuple[float, float] = slider("foo", 5.0, 10.0)
```

Comment:
Hm, shouldn't the two positional arg case return a single `int`? The two positional args should be interpreted as the min/max in this case, and we don't consider it a range slider unless there's also a tuple value provided.

I believe it's the case in general that sliders don't return tuples unless their default values are set to tuples. The exception to this is that the slider may become a range slider if its value is set to a tuple via session state, but there's probably no good way of encapsulating that case in static type checking without having to manually cast things.

(Same comment for the kwarg equivalent of these tests below / for date and time sliders too)

Reply:
Ye, I think I misread the docs/source code. Nothing we can do really about the `st.session_state` injection case.